### ISSUE TITLE
Custom cluster registration IPv6 ips are formatted incorrectly

### ISSUE THREAD

**Issue by jakefhyde** [rancher/dashboard#9660]
**Setup**
- Rancher version: v2.7-head
- Rancher UI Extensions: N/A
- Browser type & version: Google Chrome Version 116.0.5845.140 (Official Build) unknown (64-bit)

**Describe the bug**

When entering an IPv6 address for custom cluster registration, the UI drops the `:` character, resultin in an incorrectly formatted IPv6 address.

**To Reproduce**

- Create a custom rke2/k3s cluster
- Navigate to the registration tab and click advanced
- Attempt to set the public or private ip to an IPv6 address, such as `2001:db8:3333:4444:5555:6666:7777:8888`.

**Result**

The following flag is added to the registration command:
```sh
--address 2001db8333344445555666677778888
```

**Expected Result**

```sh
--address '2001:db8:3333:4444:5555:6666:7777:8888'
```

**Screenshots**

![Screenshot from 2023-09-06 14-00-33](https://github.com/rancher/dashboard/assets/33796120/984631d9-2f30-465e-b008-e33eb753af9c)

**Additional context**

Found while investigating https://github.com/rancher/rancher/issues/42411

**Comment by markusewalker** [comment]
**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1|As a standard user, attempt to provision a custom cluster with IPv6 specified and validate that the registration command correctly formats IPv6|

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.8-head` https://github.com/rancher/rancher/commit/a9ea4ce4a5bab9dd112cd20c9e250aba4dbe5176.

**ENVIRONMENT DETAILS**
- Rancher version: `v2.8-head` https://github.com/rancher/rancher/commit/a9ea4ce4a5bab9dd112cd20c9e250aba4dbe5176
- Rancher install: Docker

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, attempt to provision a custom cluster with IPv6 specified and validate that the registration command correctly formats IPv6|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As a standard user, attempted to provision a downstream RKE2 `v1.27.5+rke2r1` custom cluster.
2. In the `Advanced` section, specified the IPv6 address in the `Node Public IP` address; validated that the registration command correctly parses the `--address` flag.
3. Unable to successfully properly provision due to known issue https://github.com/rancher/rancher/issues/42411.

**Comment by markusewalker** [comment]
Reopening this issue. On further testing https://github.com/rancher/rancher/issues/42411, I neglected to test if the private IPv6 address is formatted correctly...it is not. It has the same issue as the issue description states.
